src/standard/TIP-4/6.md

```diff
@@ -0,0 +1,184 @@
+---
+title: 4.6. Upgradeable NFT
+sidebar_position: 6
+slug: /standard/TIP-4.6
+---
+
+# Upgradeable NFT (TIP-4.6)
+Requires: [TIP-4.1](1.md)
+Requires: [TIP-6.1](./../TIP-6/1.md)
+
+## Abstract
+
+This standard describes the operation of upgradeable NFT contracts. This is based on [TIP-4.1](1.md) and does not describe the functionality proposed there. 
+
+The only difference in the minting process is that the Collection deploys an NftPlatform contract rather than an NFT.
+
+Immediately after deployment to the network, the NftPlatform contract calls the `tvm.setcode()`, `tvm.setCurrentCode()`, and `onCodeUpgrade` functions, changing its code to NFT code.
+
+## Motivation
+
+The standard allows the NFT code to be changed in case an error is found in it or there is a need to add new functionality.
+
+## Specification
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+
+## Collection
+
+Collection stores the current NFT code and its version, and upgrades its code to the current code upon request from the NFT.
+
+Code update functions are not standardized.
+
+```solidity
+interface ITIP4_6Collection {
+    event UpgradeNftRequested(uint32 oldVersion, uint32 newVersion, address nft, address initiator);
+    event NftCodeUpdated(uint32 oldVersion, uint32 newVersion, uint256 oldCodeHash, uint256 newCodeHash);
+
+    function nftVersion() external view responsible returns (uint32);
+	
+    function platformCode() external view responsible returns (TvmCell);
+	
+    function platformCodeInfo() external view responsible returns (uint256 codeHash, uint16 codeDepth);
+
+    function gasUpgradeValue() external view responsible returns (uint128 fixedValue, uint128 dynamicValue);
+}
+```
+
+### TIP4_6Collection.nftVersion()
+```solidity
+function nftVersion() external view responsible returns (uint32);
+```
+
+Returns the current version of the NFT. The Collection stores only the latest version, update history is stored in events.
+
+### TIP4_6Collection.platformCode()
+```solidity
+function nftVersion() external view responsible returns (uint32);
+```
+
+Returns the NftPlatform code containing the `collection`(`address`) salt. The NftPlatform code is non-upgradeable.
+
+### TIP4_6Collection.platformCodeInfo()
+```solidity
+function platformCodeInfo() external view responsible returns (uint256 codeHash, uint16 codeDepth);
+```
+
+Returns the hash and depth of the NftPlatform code. These can be used to calculate the NFT address in third-party contracts. The hash is taken from the NftPlatform code containing the `collection`(`address`) salt.
```

Comment:
If you explain more (how to calculate NFT address in third-party contracts to use codeHash and codeDepth